ci/scripts/wait-for-other-builds-to-finish.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+: ${CIRCLECI_API_KEY:?"Required Env Variable not found!"}
+: ${CIRCLE_BUILD_NUM:?"Required Env Variable not found!"}
+: ${CIRCLE_PROJECT_USERNAME:?"Required Env Variable not found!"}
+: ${CIRCLE_PROJECT_REPONAME:?"Required Env Variable not found!"}
+: ${CIRCLE_REPOSITORY_URL:?"Required Env Variable not found!"}
+
+if [ -z "$1" ]; then
+	echo "Must provide Max Queue Time in *minutes* as script argument"
+	exit 1
+fi
+
+readonly MAX_TIME=$1
+
+wait_for_previous_builds_to_complete(){
+	local max_time_seconds=$((MAX_TIME * 60))
+
+	until [[ $(curl -s "https://circleci.com/api/v1.1/project/github/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}?circle-token=${CIRCLECI_API_KEY}&filter=running" \
+		| jq 'min_by(.build_num) | .build_num') \
+		-eq ${CIRCLE_BUILD_NUM} ]]; do
+
+			sleep 10
+			printf "."
+			if [[ "${SECONDS}" -gt "${max_time_seconds}" ]]; then
+				printf "Max time exceeded waiting to reach front of queue, cancelling build"
+				cancel_current_build
+				sleep 20
```

Comment:
why do we have to sleep here?

Reply:
Just to give a chance for the build to be cancelled gracefully otherwise it would show as failed. I can reduce the wait time and add a comment

Reply:
I am not concerned about the time, but explaining the reason would be great. Thanks!